src/agent/debuglet.ts

```diff
@@ -242,6 +243,15 @@ export class Debuglet extends EventEmitter {
     this.breakpointFetched = null;
     this.breakpointFetchedTimestamp = -Infinity;
     this.debuggeeRegistered = new CachedPromise();
+
+    const that = this;
```

Comment:
It'll be better to use a fat arrow function and remove this `that = this` hack.

```typescript
const debugAgentWarningListener = (warning: ...) => {
  ...
};
```

But I'm not sure `fn.name` in the test is still `debugAgentWarningListener`.

Reply:
It still works with arrow functions.  I've updated the code.

Reply:
Great. I like this better than the original one. BTW now that I look at the code again, why not just make `warningListener` or `debugAgentWarningListener` a private member function?